src/compiler/scala/tools/nsc/backend/jvm/BCodeSkelBuilder.scala

```diff
@@ -112,10 +120,30 @@ abstract class BCodeSkelBuilder extends BCodeHelpers {
       val optSerial: Option[Long] = serialVUID(claszSymbol)
       if (optSerial.isDefined) { addSerialVUID(optSerial.get, cnode)}
 
-      addClassFields()
-
-      gen(cd.impl)
-
+      if (cd.symbol.hasJavaAnnotationFlag) {
+        cd.impl.filter(_.symbol == null).foreach(t => warning(s"$t: symbol is null!"))
+        val constr = exitingTyper { cd.symbol.primaryConstructor }.asMethod // doesn't get us any further ...
+        warning(s"${constr.firstParam.isValueParameter}, ${constr.firstParam.tpe}")
+        cd.impl.find(tree => tree.symbol != null &&
+            tree.symbol.isConstructor)
+            .map {
+          tree =>
+            val cstr = tree.asInstanceOf[DefDef]
+            val params = cstr.vparamss.head
+            params.foreach { param =>
+              if (param.symbol.hasDefault) {
+                val default = param.rhs // <-- <empty>
```

Comment:
@xeno-by Well, I can see that the data is there after typer and I could still access the `defaultX` field in the companion object.
But it can't be _that_ hard to return something scalac already has, right?